File: server/Models/Vault.cs
using System.ComponentModel.DataAnnotations;

namespace keepr.Models;

public class Vault : DbItem<int>
{
    public string Name { get; set; }
    public string CreatorId { get; set; }
    public string Description { get; set; }

    [Url]
    public string Img { get; set; }
    public bool? IsPrivate { get; set; }
    public Profile Creator { get; set; }
}

File: client/src/models/Vault.js
import { Account } from "./Account.js";

export class Vault {
	constructor(data) {
		this.id = data.id;
		this.name = data.name;
		this.description = data.description;
		this.creator = new Account(data.creator);
		this.creatorId = data.creatorId;
		this.coverImg = data.coverImg;
		this.createdAt = data.createdAt;
		this.updatedAt = data.updatedAt;
		this.isPrivate = data.isPrivate;
		this.views = data.views;
		this.kept = data.kept;
	}
}
